app/src/main/java/com/example/shiv/mdbpokedexproject2/PokedexAdapter.java

```diff
@@ -0,0 +1,158 @@
+package com.example.shiv.mdbpokedexproject2;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.v7.util.SortedList;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.SearchView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.RatingBar;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Created by Shiv on 2/11/17.
+ */
+
+public class PokedexAdapter extends RecyclerView.Adapter<PokedexAdapter.CustomViewHolder> {
+
+
+    private final SortedList<Pokedex.Pokemon> mSortedList = new SortedList<>(Pokedex.Pokemon.class, new SortedList.Callback<Pokedex.Pokemon>() {
+        @Override
+        public int compare(Pokedex.Pokemon a, Pokedex.Pokemon b) {
+            return mComparator.compare(a, b);
+        }
+
+        @Override
+        public void onInserted(int position, int count) {
+            notifyItemRangeInserted(position, count);
+        }
+
+        @Override
+        public void onRemoved(int position, int count) {
+            notifyItemRangeRemoved(position, count);
+        }
+
+        @Override
+        public void onMoved(int fromPosition, int toPosition) {
+            notifyItemMoved(fromPosition, toPosition);
+        }
+
+        @Override
+        public void onChanged(int position, int count) {
+            notifyItemRangeChanged(position, count);
+        }
+
+        @Override
+        public boolean areContentsTheSame(Pokedex.Pokemon oldItem, Pokedex.Pokemon newItem) {
+            return oldItem.equals(newItem);
+        }
+
+        @Override
+        public boolean areItemsTheSame(Pokedex.Pokemon item1, Pokedex.Pokemon item2) {
```

Comment:
This is checking the actual references, the method right before this checks the contents